umh-core/pkg/communicator/actions/edit-dataflowcomponent.go

```diff
@@ -184,6 +187,15 @@ func (a *EditDataflowComponentAction) Parse(payload interface{}) error {
 		return errors.New("missing required field Name")
 	}
 
+	a.state = models.ComponentState(topLevel.State)
+	if a.state == "" {
+		a.state = models.ComponentStateActive
+	}
+	if !a.state.IsValid() {
+		SendActionReply(a.instanceUUID, a.userEmail, a.actionUUID, models.ActionFinishedWithFailure, "invalid state: "+string(a.state), a.outboundChannel, models.EditDataFlowComponent)
```

Comment:
why is this not V2?

Reply:
wait, wasnt there a logic like this:

1. send action replies within the function
2. but if it fails, do not send the action reply with failure but instead return the error (which then will be automatically translated into another action message?

i remember it was quite weird from the beginning